engine/src/main/scala/cromwell/engine/workflow/WorkflowStoreActor.scala

```diff
@@ -76,37 +76,35 @@ abstract class WorkflowStore {
   }
 
   def remove(id: WorkflowId): Boolean = {
-    val newWorkflowStore = workflowStore filterNot { _.id == id }
-
-    if (newWorkflowStore == workflowStore) {
-      false
-    } else {
-      workflowStore = newWorkflowStore
+    if (workflowStore.exists(_.id == id)) {
```

Comment:
Don't really agree w/ this change although since this logic will be db-ized imminently I suppose it doesn't matter either way